### PR TITLE
`azurerm_media_streaming_endpoint` - correctly support `tags`, `current_sku`, and `scale_units`

### PR DESCRIPTION
**Pull request by teowa**
- fix `tags`
- export `current_sku` 
- change `scale_units` maxmium to `50`
- refactor to retrieve the existing and use `d.HasChanges` consistently in Update

Test:
```
make acctests SERVICE="media" TESTARGS="-parallel 50 -test.run=TestAccMediaStreamingEndpoint" TESTTIMEOUT='1440m' 
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/media -parallel 50 -test.run=TestAccMediaStreamingEndpoint -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccMediaStreamingEndpoint_basic
=== PAUSE TestAccMediaStreamingEndpoint_basic
=== RUN   TestAccMediaStreamingEndpoint_CDN
=== PAUSE TestAccMediaStreamingEndpoint_CDN
=== RUN   TestAccMediaStreamingEndpoint_MaxCacheAge
=== PAUSE TestAccMediaStreamingEndpoint_MaxCacheAge
=== RUN   TestAccMediaStreamingEndpoint_shouldStopWhenStarted
=== PAUSE TestAccMediaStreamingEndpoint_shouldStopWhenStarted
=== RUN   TestAccMediaStreamingEndpoint_update
=== PAUSE TestAccMediaStreamingEndpoint_update
=== RUN   TestAccMediaStreamingEndpoint_standard
=== PAUSE TestAccMediaStreamingEndpoint_standard
=== CONT  TestAccMediaStreamingEndpoint_basic
=== CONT  TestAccMediaStreamingEndpoint_shouldStopWhenStarted
=== CONT  TestAccMediaStreamingEndpoint_standard
=== CONT  TestAccMediaStreamingEndpoint_MaxCacheAge
=== CONT  TestAccMediaStreamingEndpoint_CDN
=== CONT  TestAccMediaStreamingEndpoint_update
--- PASS: TestAccMediaStreamingEndpoint_standard (229.54s)
--- PASS: TestAccMediaStreamingEndpoint_MaxCacheAge (230.36s)
--- PASS: TestAccMediaStreamingEndpoint_CDN (234.82s)
--- PASS: TestAccMediaStreamingEndpoint_basic (236.53s)
--- PASS: TestAccMediaStreamingEndpoint_shouldStopWhenStarted (247.35s)
--- PASS: TestAccMediaStreamingEndpoint_update (325.54s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/media 325.588s

```

### CODE REVIEW

File: internal/services/media/media_streaming_endpoint_resource.go [hunk 1]
```diff
@@ -322,44 +331,57 @@ func resourceMediaStreamingEndpointUpdate(d *pluginsdk.ResourceData, meta interf
 		}
 	}
 
-	if _, ok := d.GetOk("access_control"); ok {
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	if resp.Model == nil || resp.Model.Properties == nil {
+		return fmt.Errorf("unexpected null model of %s", *id)
```
Comment (stephybun) [inline internal/services/media/media_streaming_endpoint_resource.go:339]:
```suggestion
		return fmt.Errorf("retrieving %s: model was nil", *id)
```

File: internal/services/media/media_streaming_endpoint_resource.go [hunk 2]
```diff
@@ -137,6 +137,23 @@ func resourceMediaStreamingEndpoint() *pluginsdk.Resource {
 				},
 			},
 
+			"current_sku": {
```
Comment (stephybun) [inline internal/services/media/media_streaming_endpoint_resource.go:140]: Can the sku not be set?

Comment (teowa) [reply internal/services/media/media_streaming_endpoint_resource.go:140]: image

Comment (teowa) [reply internal/services/media/media_streaming_endpoint_resource.go]: changed to `sku`

**Review by stephybun — APPROVED** [review]
Thanks @teowa LGTM 👍

**Pull request merged**